src/hotspot/share/opto/addnode.cpp

```diff
@@ -1004,14 +991,29 @@ const Type *XorINode::add_ring( const Type *t0, const Type *t1 ) const {
   const TypeInt *r0 = t0->is_int(); // Handy access
   const TypeInt *r1 = t1->is_int();
 
-  // Complementing a boolean?
-  if( r0 == TypeInt::BOOL && ( r1 == TypeInt::ONE
-                               || r1 == TypeInt::BOOL))
-    return TypeInt::BOOL;
+  if( !r0->is_con() || !r1->is_con() ) {
```

Comment:
```suggestion
  if( !r0->is_con() || !r1->is_con()) {
```
That is the new style, please apply it everywhere in the code.